components/wifi/esp8266-driver/ESP8266/ESP8266.cpp

```diff
@@ -550,6 +548,9 @@ bool ESP8266::dns_lookup(const char *name, char *ip)
 
 nsapi_error_t ESP8266::send(int id, const void *data, uint32_t amount)
 {
+    // +CIPSEND supports up to 2048 bytes at a time
+    amount = amount > 2048 ? 2048 : amount;
```

Comment:
I think we need check here if this is TCP or UDP.

For TCP this is fine.
For UDP this needs to return parameter error.